File: year2/semester2/mpp/c#/TripClientServer/TripCommon/Dto/TripDto.cs
using System;

namespace TripCommon.Dto
{
    [Serializable]
    public class TripDto
    {
        public int id;
        public string landmark;
        public string companyName;
        public DateTime departure;
        public float price;
        public int places;
        public int start;
        public int end;

        public TripDto(int tripId)
        {
            this.id = tripId;
        }

        public TripDto(string landmark, int start, int end)
        {
            this.landmark = landmark;
            this.start = start;
            this.end = end;
        }

        public TripDto(int id, string landmark, string companyName, DateTime departure, float price, int places)
        {
            this.id = id;
            this.landmark = landmark;
            this.companyName = companyName;
            this.departure = departure;
            this.price = price;
            this.places = places;
        }
    }
}

File: year2/semester2/mpp/java/TripClientServer/TripCommon/src/main/java/dto/TripDto.java
package dto;

import jdk.internal.net.http.common.Pair;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.ArrayList;

public class TripDto implements Serializable {
    public Integer id;
    public String landmark;
    public String companyName;
    public LocalDateTime departureTime;
    public float price;
    public int availablePlaces;
    public ArrayList<Integer> interval;

    public TripDto(String landmark, ArrayList<Integer> interval) {
        this.landmark = landmark;
        this.interval = interval;
    }

    public TripDto(Integer id, String landmark, String companyName, LocalDateTime departureTime, float price, int availablePlaces) {
        this.id = id;
        this.landmark = landmark;
        this.companyName = companyName;
        this.departureTime = departureTime;
        this.price = price;
        this.availablePlaces = availablePlaces;
    }

    public TripDto(Integer tripId) {
        id = tripId;
    }

    @Override
    public String toString() {
        return "TripDto{" +
                "id=" + id +
                ", landmark='" + landmark + '\'' +
                ", companyName='" + companyName + '\'' +
                ", departureTime=" + departureTime +
                ", price=" + price +
                ", availablePlaces=" + availablePlaces +
                '}';
    }
}
